Makefile

```diff
@@ -51,43 +49,42 @@ all: clean
 	fi
 	# XXX: This is a hack that we can hopefully get rid of once. Currently
 	# the livefs Launchpad builders don't have multiverse enabled.
-	# We wanto to work-around that by actually enabling multiverse just
+	# We want to work-around that by actually enabling multiverse just
 	# for this one build here as we need it for linux-firmware-raspi2.
 	$(call enable_multiverse)
+
 	# Preparation stage
 	mkdir -p $(STAGEDIR)/debs $(STAGEDIR)/unpack
-	# u-boot
+	$(call stage_package,flash-kernel,$(STAGEDIR),$(ARCH))
```

Comment:
Good to see we'll be finally pulling the bootscript from a common place (flash-kernel). Just a note here: before we can actually merge this, we'll probably need to SRU flash-kernel to bionic, since currently it doesn't have the bootscr.rpi in it. Even though the next bionic point release is still in some time, we do build dailies for that series. So I'd say we'd want flash-kernel in at least bionic-proposed.

Reply:
Actually I don't think this is an issue. There's a very subtle change in the script which deals with this different. Specifically, Eoan (and Disco) have bootscript/all/bootscr.rpi in flash-kernel, while Bionic has bootscript/all/bootscr.rpi3 (and a uboot.env for pi2) in its flash-kernel. Now, on line 66 note that mkimage is called with `.../bootscr.rpi*` rather than `.../bootscr.rpi` to account for this difference (yes, it's a bit hacky to use a wildcard here but I think it's acceptable given the requirement).

Incidentally, this is (one of the reasons) why I suggested going through the individual commits rather than just looking at the end state; 17596c4f3458b4e368224453f1defe0ce2bfaa69 mentions this case in the commit message.

Reply:
hm, you are right and flash-kernel from bionic does have the base bootscript for the pi3. But I still think there will be a problem right now - a bit of a different one though. So the builds will not FTBFS, but I suppose our arm64 bionic daily images will stop working.

The old flash-kernel bootscr.rpi3 that we have there I guess is very armhf oriented (which is why it's in 'armhf' instead of 'all'), supporting only booting the armhf kernel - since there are no conditional checks there for now. Certainly this is less of a blocker, as we never really give any guarantee of bootability of our daily images, but it's something we should sort out earlier before the point release. So I'd say let's aim for getting it into bionic-proposed at least in the nearest time.